### PR TITLE
[SIG-4763] file PreviousButton has been changed to typescript.

### PR DESCRIPTION
**Pull request by SireeKoolenWijkstra**
File PreviousButton has been changed to typescript.

### CODE REVIEW

File: src/components/PreviousButton/index.tsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
 // SPDX-License-Identifier: MPL-2.0
 // Copyright (C) 2020 - 2021 Gemeente Amsterdam
-import PropTypes from 'prop-types'
 import styled from 'styled-components'
 import { ChevronLeft } from '@amsterdam/asc-assets'
 import { themeColor } from '@amsterdam/asc-ui'
 
 import Button from 'components/Button'
+import type { FunctionComponent } from 'react'
```
Comment (OscarBakker) [inline src/components/PreviousButton/index.tsx:8]: please move this line to the top, ordering stuff

File: src/components/PreviousButton/index.tsx [hunk 2]
```diff
@@ -16,7 +16,17 @@ const Chevron = styled(ChevronLeft)`
   fill: ${themeColor('primary')};
 `
 
-const PreviousButton = ({ className, children, onClick }) => (
+export interface PreviousButtonProps {
+  className?: string
+  children: string
```
Comment (OscarBakker) [inline src/components/PreviousButton/index.tsx:21]: Please use the right typing. ReactNode

**Review by OscarBakker — COMMENTED** [review]
few things

**Review by vdegraaf — COMMENTED** [review]
Maybe good to change the file name as well too previousButton.tsx and export it through the index.ts.

File: src/components/PreviousButton/index.tsx (hunk 1 above)
Comment (SireeKoolenWijkstra) [reply src/components/PreviousButton/index.tsx:8]: Your remarks has been resolved (as in: done and done)

File: src/components/PreviousButton/index.tsx [hunk 3]
```diff
@@ -1,11 +1,11 @@
 // SPDX-License-Identifier: MPL-2.0
 // Copyright (C) 2020 - 2021 Gemeente Amsterdam
```
[the remark this answers is not in the document]
Comment (SireeKoolenWijkstra) [reply src/components/PreviousButton/index.tsx:2]: Your remarks has been resolved (as in: done, done and done)

File: docs/adr/0006-forms.md [hunk 4]
```diff
@@ -58,11 +58,11 @@ These libraries can exist side by side in the code base. New forms can be built
 
 ### Incident form
 
-- [IncidentForm/index.js](../../src/signals/incident/components/IncidentForm/index.js)
+- [IncidentForm/PreviousButton.tsx](../../src/signals/incident/components/IncidentForm/index.js)
```
Comment (kiruys) [inline docs/adr/0006-forms.md:61]: is this url still correct?

File: docs/adr/0006-forms.md [hunk 5]
```diff
@@ -58,11 +58,11 @@ These libraries can exist side by side in the code base. New forms can be built
...
 
 A form components specifically using the dependency, which can easily be updated when using the new lib.
 
-- [form/Header/index.js](../../src/signals/incident/components/form/Header/index.js)
+- [form/Header/PreviousButton.tsx](../../src/signals/incident/components/form/Header/index.js)
```
Comment (kiruys) [inline docs/adr/0006-forms.md:65]: same here

Comment (kiruys) [reply docs/adr/0006-forms.md:65]: and below

File: src/components/PreviousButton/PreviousButton.tsx [hunk 6]
```diff
@@ -16,7 +15,17 @@ const Chevron = styled(ChevronLeft)`
   fill: ${themeColor('primary')};
 `
 
-const PreviousButton = ({ className, children, onClick }) => (
+export interface PreviousButtonProps {
```
Comment (OscarBakker) [inline src/components/PreviousButton/PreviousButton.tsx:18]: - [ ] no need export interface. export can be remove cause its only needed here.

File: src/components/PreviousButton/index.tsx [hunk 7]
```diff
@@ -16,7 +16,17 @@ const Chevron = styled(ChevronLeft)`
   fill: ${themeColor('primary')};
 `
 
-const PreviousButton = ({ className, children, onClick }) => (
+export interface PreviousButtonProps {
+  className?: string
```
[the remark this answers is not in the document]
Comment (OscarBakker) [reply src/components/PreviousButton/index.tsx]: by default it seems an empty string, so we could add a default empty string but we can keep it nullable.

[the remark this answers is not in the document]
Comment (OscarBakker) [reply src/components/PreviousButton/index.tsx]: can be removed for this use case... only used once

**Review by OscarBakker — APPROVED** [review]
lgtm!

**Pull request merged**